frontend/src/components/ui/question/QuestionBox.js

```diff
@@ -82,7 +98,9 @@ function QuestionBox() {
                 <QuestionTitle title={title} />
               </Grid>
               <Grid item>
-                <Button className={difficultyButtonStyle(difficulty)} disabled>{difficulty}</Button>
+                <Button className={difficultyButtonStyle(difficulty)} disabled>
```

Comment:
I think should check if `difficulty` is null/undefined before passing.

Reply:
Similar checking applies to `categories`

Reply:
Actually, I'm curious why we still have to perform a null check, since we already did a check in Match index.js, thus there is never a scenario when difficulty will be null? Please advice on this.
![image](https://user-images.githubusercontent.com/56862300/198818783-cd9c9c6f-f3cf-4e0c-851f-9f0dcfc72d13.png)

Reply:
oh right, I missed out on that, my bad. Yea if that's the case I think it's ok. But we can't eliminate the case that we might refactor our code and somehow the check is not done in matching anymore. So, it's safer to do in every layer. But due to the size of this project, we might not make alot of changes so I think it's fine.